plugins/woocommerce-blocks/assets/js/blocks/mini-cart/block.tsx

```diff
@@ -151,16 +151,30 @@ const MiniCartBlock = ( attributes: Props ): JSX.Element => {
 					selector: '.wp-block-woocommerce-mini-cart-contents',
 					blockMap: getRegisteredBlockComponents( blockName ),
 				} );
+				rootRef.current = renderedBlock;
 			}
 		}
 
 		return () => {
 			if ( contentsNode instanceof Element && isOpen ) {
-				const container = contentsNode.querySelector(
+				const unmountingContainer = contentsNode.querySelector(
 					'.wp-block-woocommerce-mini-cart-contents'
 				);
-				if ( container ) {
-					unmountComponentAtNode( container );
+
+				if ( unmountingContainer ) {
+					const foundRoot = rootRef?.current?.find(
+						( { container } ) => unmountingContainer === container
+					);
+					if (
+						typeof foundRoot === 'undefined' ||
+						! isObject( foundRoot ) ||
+						typeof foundRoot?.root?.unmount !== 'function'
+					) {
+						return;
```

Comment:
I don't really like this early return style of coding, it adds more code than necessary, and I (possibly just personal opinion) always find there is additional cognitive load in finding early returns that shortcut the function. I'd say this is a nitpick except what cements this one for me is that checking if its a function then calling setTimeout would make this code way simpler.

instead we can do:

```
if ( typeof foundRoot?.root?.unmount === 'function' ) {
  setTimeout( () => {
	foundRoot.root.unmount();
	} );
}
```

Reply:
I see what you mean. I've adjusted this part based on your suggestion.